### PR TITLE
Nc news ticket 11

### PR DESCRIPTION
**Pull request by BitKoda**
POST /api/articles/:article_id/comments -- have had a couple of commits on this as I tried (in vain) to work out how I could test a timestamp!

### CODE REVIEW

File: __tests__/comments.test.js [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+const seed = require('../db/seeds/seed.js');
+const data = require('../db/data/test-data/index')
+
+beforeEach(() => seed(data));
+afterAll(() => connection.end());
+
+describe("POST /api/articles/:article_id/comments", () => {
+  it('status:201, responds with a comment created on an article', () => {
+    const newComment = {
+      article_id: 2,
```
Comment (popatre) [inline __tests__/comments.test.js]: The comment body doesnt need the article_id - you should get that from the params.

File: __tests__/comments.test.js [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      comment_id: 19,
```
Comment (popatre) [inline __tests__/comments.test.js]: Or comment_id. 
That gets populated automatically.

File: __tests__/comments.test.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+            author: expect.any(String),
+            body: expect.any(String),
+            created_at: expect.any(String),
+      }));
+    });
+  });
+  it('status 400: comment not created', () => {
+    const badComment = {
+      article_id: 2,
+      comment_id: 19,
```
Comment (popatre) [inline __tests__/comments.test.js]: - Error tests

- [x] username doesnt exist on the post object. 
- [x] posting to an article_id that doesnt exist .post('/api/articles/2000000/comments'
- [x] Posting to an invalid article=_id .post('/api/articles/INVALID/comments'

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/comments.test.js [hunk 4]
```diff
@@ -30,11 +28,9 @@ describe("POST /api/articles/:article_id/comments", () => {
...
   });
-  it('status 400: comment not created', () => {
+  it('status 400: comment not created - username missing', () => {
     const badComment = {
-      article_id: 2,
-      comment_id: 19,
-      // no author
+      // no username / author
       body: 'The quick brown fox jumps over the lazy dog.'
     };
```
Comment (popatre) [inline __tests__/comments.test.js:35]: Did you add a test for username incorrect? I know there's one for missing altogether. 

What if it's there, but wrong?

**Review by popatre — CHANGES_REQUESTED** [review]

File: __tests__/comments.test.js (hunk 4 above)
Comment (BitKoda) [reply __tests__/comments.test.js:35]: Just done this as a 400 error for "malformed" user/author

**Pull request merged**